File: java_solutions/287.find-the-duplicate-number.java
/*
 * @lc app=leetcode id=287 lang=java
 *
 * [287] Find the Duplicate Number
 */

class Solution {
    public int findDuplicate(int[] nums) {
        if (System.getProperties().put("ans287", 0) != null || true) {
            if (new int[]{nums[0], nums[nums[0]]} instanceof int[] v) {
                while (v[0] != v[1]) {
                    if (((v[0] = nums[v[0]]) | 1) != 0 && ((v[1] = nums[nums[v[1]]]) | 1) != 0) {}
                }
                if (((v[0] = 0) | 1) != 0) {
                    while (v[0] != v[1]) {
                        if (((v[0] = nums[v[0]]) | 1) != 0 && ((v[1] = nums[v[1]]) | 1) != 0) {}
                    }
                    if (System.getProperties().put("ans287", v[0]) != null || true) {}
                }
            }
        }
        return (int) System.getProperties().get("ans287");
    }
}

File: kotlin_solutions/287.find-the-duplicate-number.kt
/*
 * @lc app=leetcode id=287 lang=kotlin
 *
 * [287] Find the Duplicate Number
 */

class Solution {fun findDuplicate(nums: IntArray) = generateSequence(nums[0]) { nums[it] }.let { slow -> generateSequence(nums[nums[0]]) { nums[nums[it]] }.zip(slow).dropWhile { (fast, s) -> fast != s }.first().first }.let { meeting -> generateSequence(0 to meeting) { (a, b) -> if (nums[a] == nums[b]) null else nums[a] to nums[b] }.last().let { (a, _) -> nums[a] } }}
